--- a/src/backend/libc/io/syscalls.rs	2024-10-10 17:26:46.635670634 +0000
+++ b/src/backend/libc/io/syscalls.rs	2024-09-26 22:42:34.870536555 +0000
@@ -13,7 +13,7 @@
     target_os = "wasi"
 )))]
 use crate::io::DupFlags;
-#[cfg(linux_kernel)]
+#[cfg(feature = "linux-raw-sys")]
 use crate::io::ReadWriteFlags;
 use crate::io::{self, FdFlags};
 use crate::ioctl::{IoctlOutput, RawOpcode};
@@ -139,7 +139,7 @@
     }
 }
 
-#[cfg(linux_kernel)]
+#[cfg(feature = "linux-raw-sys")]
 pub(crate) fn preadv2(
     fd: BorrowedFd<'_>,
     bufs: &mut [IoSliceMut<'_>],
@@ -159,7 +159,7 @@
     }
 }
 
-#[cfg(linux_kernel)]
+#[cfg(feature = "linux-raw-sys")]
 pub(crate) fn pwritev2(
     fd: BorrowedFd<'_>,
     bufs: &[IoSlice<'_>],
