
---
 src/lib.rs | 4 ++++
 1 file changed, 4 insertions(+)

diff --git b/src/lib.rs a/src/lib.rs
index 220af6ba..cd5cca0b 100644
--- b/src/lib.rs
+++ a/src/lib.rs
@@ -247,6 +247,10 @@ appropriate `repr` attribute:
 #![warn(clippy::all)]
 #![no_std]
 
+/// Local Android change: Use std to allow building as a dylib.
+#[cfg(android_dylib)]
+extern crate std;
+
 use core::fmt::{Debug, Formatter, Result};
 use core::ops::*;
 
