diff --git a/Android.bp b/Android.bp
index 15ca6c0..c0c66d4 100644
--- a/Android.bp
+++ b/Android.bp
@@ -19,21 +19,7 @@ rust_library {
     cargo_env_compat: true,
     cargo_pkg_version: "1.6.2+v1.6.2",
     crate_root: "src/lib.rs",
-    edition: "2021",
-    features: ["bindgen"],
-    apex_available: [
-        "//apex_available:platform",
-        "//apex_available:anyapex",
-    ],
-    min_sdk_version: "29",
-}
-
-rust_ffi_static {
-    name: "liblibbpf_sys_static",
-    crate_name: "libbpf_sys",
-    cargo_env_compat: true,
-    cargo_pkg_version: "1.6.2+v1.6.2",
-    crate_root: "src/lib.rs",
+    srcs: [":liblibbpf_bindgen"],
     edition: "2021",
     features: ["bindgen"],
     apex_available: [
