diff --git a/Android.bp b/Android.bp
index 745d2e8..8e44ff7 100644
--- a/Android.bp
+++ b/Android.bp
@@ -21,6 +21,14 @@ rust_library {
     cargo_pkg_version: "0.8.12",
     crate_root: "src/lib.rs",
     edition: "2018",
+    arch: {
+        arm64: {
+            features: ["folded_multiply"],
+        },
+        x86_64: {
+            features: ["folded_multiply"],
+        },
+    },
     features: [
         "default",
         "getrandom",
@@ -28,7 +36,6 @@ rust_library {
         "std",
     ],
     cfgs: [
-        "folded_multiply",
         "specialize",
     ],
     rustlibs: [
