
---
 tests/test_bytes.rs | 1 +
 1 file changed, 1 insertion(+)

diff --git b/tests/test_bytes.rs a/tests/test_bytes.rs
index 5ec60a5b..da1b0743 100644
--- b/tests/test_bytes.rs
+++ a/tests/test_bytes.rs
@@ -288,6 +288,7 @@ fn split_to_uninitialized() {
 }
 
 #[test]
+#[ignore = "Android: we unwind differently."]
 fn split_off_to_at_gt_len() {
     fn make_bytes() -> Bytes {
         let mut bytes = BytesMut::with_capacity(100);
