diff --git a/src/definition.rs b/src/definition.rs
index f53fee698..dcdabceed 100644
--- a/src/definition.rs
+++ b/src/definition.rs
@@ -28,9 +28,7 @@ pub mod layout;
 pub mod modifier;
 pub mod view;
 
-// LINT.IfChange
 pub static CURRENT_VERSION: u32 = 27;
-// Lint.ThenChange(common/src/main/java/com/android/designcompose/common/DCDVersion.kt)
 
 impl DesignComposeDefinitionHeader {
     pub fn current(
