
---
 src/lib.rs | 4 ++++
 1 file changed, 4 insertions(+)

diff --git b/src/lib.rs a/src/lib.rs
index dcd5097d..0bb32c92 100644
--- b/src/lib.rs
+++ a/src/lib.rs
@@ -320,6 +320,10 @@
 //! [`UtcTime`]: asn1::UtcTime
 //! [`Utf8StringRef`]: asn1::Utf8StringRef
 
+/// Local Android change: Use std to allow building as a dylib.
+#[cfg(android_dylib)]
+extern crate std;
+
 #[cfg(feature = "alloc")]
 #[allow(unused_imports)]
 #[macro_use]
