diff --git a/Android.bp b/Android.bp
index 8d77a6276..f03a63f4d 100644
--- a/Android.bp
+++ b/Android.bp
@@ -55,8 +55,8 @@ rust_test_host {
 }
 
 rust_binary_host {
-    name: "protoc_gen_rs",
-    crate_name: "protoc_gen_rs",
+    name: "protoc-gen-rust",
+    crate_name: "protoc_gen_rust",
     cargo_env_compat: true,
     cargo_pkg_version: "3.7.2",
     crate_root: "src/bin/protoc-gen-rs.rs",
