diff --git a/crates/vhost-user-backend/src/vring.rs b/crates/vhost-user-backend/src/vring.rs
index 948f6878..0855dc89 100644
--- a/crates/vhost-user-backend/src/vring.rs
+++ b/crates/vhost-user-backend/src/vring.rs
@@ -61,6 +61,9 @@ pub trait VringT<M: GuestAddressSpace>:
     /// Set vring enabled state.
     fn set_enabled(&self, enabled: bool);
 
+    /// Get vring enabled state.
+    fn get_enabled(&self) -> bool;
+    
     /// Set queue addresses for descriptor table, available ring and used ring.
     fn set_queue_info(
         &self,
@@ -174,6 +177,10 @@ impl<M: GuestAddressSpace> VringState<M> {
         self.enabled = enabled;
     }
 
+    pub fn get_enabled(&self) -> bool {
+	self.enabled
+    }
+    
     /// Set queue addresses for descriptor table, available ring and used ring.
     pub fn set_queue_info(
         &mut self,
@@ -327,6 +334,10 @@ impl<M: 'static + GuestAddressSpace> VringT<M> for VringMutex<M> {
         self.lock().set_enabled(enabled)
     }
 
+    fn get_enabled(&self) -> bool {
+	self.get_ref().enabled
+    }
+    
     fn set_queue_info(
         &self,
         desc_table: u64,
@@ -442,6 +453,10 @@ impl<M: 'static + GuestAddressSpace> VringT<M> for VringRwLock<M> {
         self.write_lock().set_enabled(enabled)
     }
 
+    fn get_enabled(&self) -> bool {
+	self.get_ref().enabled
+    }
+
     fn set_queue_info(
         &self,
         desc_table: u64,
