diff --git a/src/debug.rs b/src/debug.rs
index edc13d0c2..1c8eeadfc 100644
--- a/src/debug.rs
+++ b/src/debug.rs
@@ -103,7 +103,8 @@ impl DumpAsCss for FlexWrap {
 impl DumpAsCss for taffy::Display {
     fn to_css_value(&self) -> String {
         match self {
-            //Display::Block => "block",
+            Display::Block => "block",
+            Display::Grid => "grid",
             Display::Flex => "flex",
             Display::None => "none",
         }
