diff --git a/src/backends/java/codegen/packet.rs b/src/backends/java/codegen/packet.rs
index 8d783c2..92978ad 100644
--- a/src/backends/java/codegen/packet.rs
+++ b/src/backends/java/codegen/packet.rs
@@ -192,6 +192,7 @@ pub fn gen_abstract_packet(
 
             $(if parent.is_none() => public abstract byte[] toBytes();)
 
+            $(if parent.is_some() => @Override)
             protected byte[] toBytes($(&*import::BB) payload) {
                 payload.rewind();
                 $(&*import::BB) buf = $(&*import::BB)
@@ -199,7 +200,7 @@ pub fn gen_abstract_packet(
                     .order($endianness);
 
                 $(encoder(&def.alignment, &def.width_fields, endianness))
-                return buf.array();
+                $(if parent.is_some() { return super.toBytes(buf); } else { return buf.array(); })
             }
 
             $(width_def(name, &ctx.heirarchy, false))
