
---
 Cargo.toml | 1 +
 src/lib.rs | 2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git b/Cargo.toml a/Cargo.toml
index 77fe68f7..bc62b4a7 100644
--- b/Cargo.toml
+++ a/Cargo.toml
@@ -10,6 +10,7 @@
 # See Cargo.toml.orig for the original contents.
 
 [package]
+edition = "2018" # ANDROID: Required to build dylib in no_std.
 name = "memoffset"
 version = "0.9.0"
 authors = ["Gilad Naaman <gilad.naaman@gmail.com>"]
diff --git b/src/lib.rs a/src/lib.rs
index ccf182b6..8d65d9aa 100644
--- b/src/lib.rs
+++ a/src/lib.rs
@@ -55,7 +55,7 @@
 //! let checksum = crc16(checksum_range);
 //! ```
 
-#![no_std]
+#![cfg_attr(not(aosp_force_use_std), no_std)]
 #![cfg_attr(
     all(feature = "unstable_const", not(stable_const)),
     feature(const_ptr_offset_from)
