diff --git a/src/crypto/boringssl.rs b/src/crypto/boringssl.rs
index 98d45d771..bdbd11533 100644
--- a/src/crypto/boringssl.rs
+++ b/src/crypto/boringssl.rs
@@ -67,7 +67,7 @@ impl PacketKey {
         derive_pkt_key(aead, secret, &mut key)?;
         derive_pkt_iv(aead, secret, &mut iv)?;
 
-        let pkt_key = Self::new(aead, key, iv, enc)?;
+        let mut pkt_key = Self::new(aead, key, iv, enc)?;
 
         // Dummy seal operation to prime the AEAD context with the nonce mask.
         //
@@ -117,7 +117,7 @@ impl PacketKey {
     }
 
     pub fn seal_with_u64_counter(
-        &self, counter: u64, ad: &[u8], buf: &mut [u8], in_len: usize,
+        &mut self, counter: u64, ad: &[u8], buf: &mut [u8], in_len: usize,
         extra_in: Option<&[u8]>,
     ) -> Result<usize> {
         let tag_len = self.alg.tag_len();
@@ -139,7 +139,7 @@ impl PacketKey {
 
         let rc = unsafe {
             EVP_AEAD_CTX_seal_scatter(
-                &self.ctx,                  // ctx
+                &mut self.ctx,              // ctx
                 buf.as_mut_ptr(),           // out
                 buf[in_len..].as_mut_ptr(), // out_tag
                 &mut out_tag_len,           // out_tag_len
@@ -350,7 +350,7 @@ extern "C" {
     ) -> c_int;
 
     fn EVP_AEAD_CTX_seal_scatter(
-        ctx: *const EVP_AEAD_CTX, out: *mut u8, out_tag: *mut u8,
+        ctx: *mut EVP_AEAD_CTX, out: *mut u8, out_tag: *mut u8,
         out_tag_len: *mut usize, max_out_tag_len: usize, nonce: *const u8,
         nonce_len: usize, inp: *const u8, in_len: usize, extra_in: *const u8,
         extra_in_len: usize, ad: *const u8, ad_len: usize,
diff --git a/src/crypto/mod.rs b/src/crypto/mod.rs
index c68446cd9..dde02cbbc 100644
--- a/src/crypto/mod.rs
+++ b/src/crypto/mod.rs
@@ -289,7 +289,7 @@ impl Seal {
     }
 
     pub fn seal_with_u64_counter(
-        &self, counter: u64, ad: &[u8], buf: &mut [u8], in_len: usize,
+        &mut self, counter: u64, ad: &[u8], buf: &mut [u8], in_len: usize,
         extra_in: Option<&[u8]>,
     ) -> Result<usize> {
         if cfg!(feature = "fuzzing") {
diff --git a/src/lib.rs b/src/lib.rs
index fe79fdee7..d1c1dbbc7 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -4964,7 +4964,7 @@ impl<F: BufFactory> Connection<F> {
         });
 
         let aead = match crypto_ctx.crypto_seal {
-            Some(ref v) => v,
+            Some(ref mut v) => v,
             None => return Err(Error::InvalidState),
         };
 
diff --git a/src/packet.rs b/src/packet.rs
index 788dc8c62..aa17b06f8 100644
--- a/src/packet.rs
+++ b/src/packet.rs
@@ -699,7 +699,7 @@ pub fn encrypt_hdr(
 
 pub fn encrypt_pkt(
     b: &mut octets::OctetsMut, pn: u64, pn_len: usize, payload_len: usize,
-    payload_offset: usize, extra_in: Option<&[u8]>, aead: &crypto::Seal,
+    payload_offset: usize, extra_in: Option<&[u8]>, aead: &mut crypto::Seal,
 ) -> Result<usize> {
     let (mut header, mut payload) = b.split_at(payload_offset)?;
 
@@ -826,7 +826,7 @@ fn compute_retry_integrity_tag(
     pb.put_bytes(odcid)?;
     pb.put_bytes(&b.buf()[..hdr_len])?;
 
-    let key = crypto::PacketKey::new(
+    let mut key = crypto::PacketKey::new(
         RETRY_AEAD_ALG,
         key.to_vec(),
         nonce.to_vec(),
@@ -1766,7 +1766,7 @@ mod tests {
 
         b.put_bytes(header).unwrap();
 
-        let (_, aead) = crypto::derive_initial_key_material(
+        let (_, mut aead) = crypto::derive_initial_key_material(
             dcid,
             hdr.version,
             is_server,
@@ -1787,7 +1787,7 @@ mod tests {
             payload_len,
             payload_offset,
             None,
-            &aead,
+            &mut aead,
         )
         .unwrap();
 
@@ -2104,7 +2104,7 @@ mod tests {
 
         let alg = crypto::Algorithm::ChaCha20_Poly1305;
 
-        let aead = crypto::Seal::from_secret(alg, &secret).unwrap();
+        let mut aead = crypto::Seal::from_secret(alg, &secret).unwrap();
 
         let pn = 654_360_564;
         let pn_len = 3;
@@ -2124,7 +2124,7 @@ mod tests {
             payload_len,
             payload_offset,
             None,
-            &aead,
+            &mut aead,
         )
         .unwrap();
 
diff --git a/src/test_utils.rs b/src/test_utils.rs
index 53e2c1d0b..27b8915d3 100644
--- a/src/test_utils.rs
+++ b/src/test_utils.rs
@@ -454,7 +454,7 @@ pub fn encode_pkt(
     }
 
     let aead = match crypto_ctx.crypto_seal {
-        Some(ref v) => v,
+        Some(ref mut v) => v,
         None => return Err(Error::InvalidState),
     };
 
diff --git a/src/tests.rs b/src/tests.rs
index bb21e1f39..40d46f985 100644
--- a/src/tests.rs
+++ b/src/tests.rs
@@ -4060,7 +4060,7 @@ fn invalid_initial_payload(
     // Use correct payload length when encrypting the packet.
     let payload_len = frames.iter().fold(0, |acc, x| acc + x.wire_len());
 
-    let aead = crypto_ctx.crypto_seal.as_ref().unwrap();
+    let aead = crypto_ctx.crypto_seal.as_mut().unwrap();
 
     let written = packet::encrypt_pkt(
         &mut b,
@@ -10368,7 +10368,7 @@ fn challenge_no_cids(
         frame.to_bytes(&mut b).expect("encode frames");
     }
 
-    let aead = crypto_ctx.crypto_seal.as_ref().expect("crypto seal");
+    let aead = crypto_ctx.crypto_seal.as_mut().expect("crypto seal");
 
     let written = packet::encrypt_pkt(
         &mut b,
