diff --git a/Android.bp b/Android.bp
index 42138df..3ad0d2c 100644
--- a/Android.bp
+++ b/Android.bp
@@ -31,31 +31,49 @@ rust_library {
         "libserde",
     ],
     proc_macros: ["libp9_wire_format_derive"],
     compile_multilib: "first",
     apex_available: [
         "//apex_available:platform",
         "//apex_available:anyapex",
     ],
     product_available: true,
     vendor_available: true,
+    // Doesn't compile on 32-bit targets.
+    target: {
+        android_arm: {
+            enabled: false,
+        },
+        android_x86: {
+            enabled: false,
+        },
+    },
 }
 
 rust_test {
     name: "p9_test_src_lib",
     host_supported: true,
     crate_name: "p9",
     cargo_env_compat: true,
     cargo_pkg_version: "0.2.3",
     crate_root: "src/lib.rs",
     test_suites: ["general-tests"],
     auto_gen_config: true,
     test_options: {
         unit_test: true,
     },
     edition: "2021",
     rustlibs: [
         "liblibc",
         "libserde",
     ],
     proc_macros: ["libp9_wire_format_derive"],
+    // Doesn't compile on 32-bit targets.
+    target: {
+        android_arm: {
+            enabled: false,
+        },
+        android_x86: {
+            enabled: false,
+        },
+    },
 }
