
---
 src/lib.rs | 3 +++
 1 file changed, 3 insertions(+)

diff --git b/src/lib.rs a/src/lib.rs
index 6ec2021e..7e46d02c 100644
--- b/src/lib.rs
+++ a/src/lib.rs
@@ -137,6 +137,9 @@ pin-project supports this.
     clippy::undocumented_unsafe_blocks,
 )]
 
+// ANDROID: Use std to allow building as a dylib.
+extern crate std;
+
 /// A macro that creates a projection type covering all the fields of struct.
 ///
 /// This macro creates a projection type according to the following rules:
