diff --git a/Android.bp b/Android.bp
index 46c34934..2e75afd8 100644
--- a/Android.bp
+++ b/Android.bp
@@ -23,8 +23,8 @@ rust_library_rlib {
     features: ["alloc"],
     rustlibs: [
         "libbitflags",
-        "liblog_rust",
-        "libzerocopy",
+        "liblog_rust_nostd",
+        "libzerocopy_nostd",
     ],
     apex_available: [
         "//apex_available:platform",
@@ -51,6 +51,6 @@ rust_test {
     rustlibs: [
         "libbitflags",
         "liblog_rust",
-        "libzerocopy",
+        "libzerocopy_nostd",
     ],
 }
