diff --git a/src/lib.rs b/src/lib.rs
index 2b4c4d79..01453fbe 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -213,12 +213,22 @@ use crate::error::Error;
 use crate::version::Version;
 use proc_macro::TokenStream;
 
-#[cfg(not(host_os = "windows"))]
+#[cfg(all(not(host_os = "windows")), not(soong))]
 const RUSTVERSION: Version = include!(concat!(env!("OUT_DIR"), "/version.expr"));
 
-#[cfg(host_os = "windows")]
+#[cfg(all(host_os = "windows"), not(soong))]
 const RUSTVERSION: Version = include!(concat!(env!("OUT_DIR"), "\\version.expr"));
 
+#[cfg(soong)]
+static RUSTVERSION: LazyLock<Version> = LazyLock::new(|| {
+    let v: Vec<&str> = ANDROID_RUSTVERSION.unwrap().split('.').collect();
+    Version {
+        minor: v[1].parse().unwrap(),
+        patch: v[2].parse().unwrap(),
+        channel: version::Channel::Stable,
+    }
+});
+
 #[proc_macro_attribute]
 pub fn stable(args: TokenStream, input: TokenStream) -> TokenStream {
     expand::cfg("stable", args, input)
