
---
 src/connection.rs | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git b/src/connection.rs a/src/connection.rs
index 53706d60..84cb8fc0 100755
--- b/src/connection.rs
+++ a/src/connection.rs
@@ -99,9 +99,9 @@ impl InnerConnection {
         hello_buf[0..BENCHMARK_MAGIC_NUMBER.len()]
             .copy_from_slice(BENCHMARK_MAGIC_NUMBER.as_bytes());
         let mut i = BENCHMARK_MAGIC_NUMBER.len();
-        hello_buf[i] = env!("CARGO_PKG_VERSION_MAJOR").parse().unwrap();
-        hello_buf[i + 1] = env!("CARGO_PKG_VERSION_MINOR").parse().unwrap();
-        hello_buf[i + 2] = env!("CARGO_PKG_VERSION_PATCH").parse().unwrap();
+        hello_buf[i] = 0;
+        hello_buf[i + 1] = 0;
+        hello_buf[i + 2] = 0;
         i += 3;
         hello_buf[i..i + 2].clone_from_slice(&PROTOCOL_VERSION.to_be_bytes());
         i += 2;
