diff --git a/src/sys/unix/uds/mod.rs b/src/sys/unix/uds/mod.rs
--- a/src/sys/unix/uds/mod.rs      2024-12-04 12:59:24.448227940 +0000
+++ b/src/sys/unix/uds/mod.rs      2024-12-04 15:02:27.953261346 +0000
@@ -158,6 +158,9 @@
     #[test]
     #[cfg(any(target_os = "android", target_os = "linux"))]
     fn abstract_address() {
+        #[cfg(target_os = "android")]
+        use std::os::android::net::SocketAddrExt;
+        #[cfg(target_os = "linux")]
         use std::os::linux::net::SocketAddrExt;

         const PATH: &[u8] = &[0, 116, 111, 107, 105, 111];
