
---
 src/lib.rs | 4 ++++
 1 file changed, 4 insertions(+)

diff --git b/src/lib.rs a/src/lib.rs
index 4e006578..1dd35f2e 100644
--- b/src/lib.rs
+++ a/src/lib.rs
@@ -235,6 +235,10 @@
 //! [good cryptographic hygiene]: https://github.com/veorq/cryptocoding#clean-memory-of-secret-data
 //! [`Ordering::SeqCst`]: core::sync::atomic::Ordering::SeqCst
 
+/// Local Android change: Use std to allow building as a dylib.
+#[cfg(android_dylib)]
+extern crate std;
+
 #[cfg(feature = "alloc")]
 extern crate alloc;
 
