diff --git a/src/fcntl.rs b/src/fcntl.rs
index cf87926c..d884cea7 100644
--- a/src/fcntl.rs
+++ b/src/fcntl.rs
@@ -223,7 +223,7 @@ pub fn open<P: ?Sized + NixPath>(
     mode: Mode,
 ) -> Result<RawFd> {
     let fd = path.with_nix_path(|cstr| unsafe {
-        libc::open(cstr.as_ptr(), oflag.bits(), mode.bits() as c_uint)
+        libc::open(cstr.as_ptr() as *const libc::c_char, oflag.bits(), mode.bits() as c_uint)
     })?;
 
     Errno::result(fd)
@@ -247,7 +247,7 @@ pub fn openat<P: ?Sized + NixPath>(
     mode: Mode,
 ) -> Result<RawFd> {
     let fd = path.with_nix_path(|cstr| unsafe {
-        libc::openat(at_rawfd(dirfd), cstr.as_ptr(), oflag.bits(), mode.bits() as c_uint)
+        libc::openat(at_rawfd(dirfd), cstr.as_ptr() as *const libc::c_char, oflag.bits(), mode.bits() as c_uint)
     })?;
     Errno::result(fd)
 }
@@ -384,9 +384,9 @@ pub fn renameat<P1: ?Sized + NixPath, P2: ?Sized + NixPath>(
         new_path.with_nix_path(|new_cstr| unsafe {
             libc::renameat(
                 at_rawfd(old_dirfd),
-                old_cstr.as_ptr(),
+                old_cstr.as_ptr() as *const libc::c_char,
                 at_rawfd(new_dirfd),
-                new_cstr.as_ptr(),
+                new_cstr.as_ptr() as *const libc::c_char,
             )
         })
     })??;
@@ -461,12 +461,12 @@ fn readlink_maybe_at<P: ?Sized + NixPath>(
             #[cfg(not(target_os = "redox"))]
             Some(dirfd) => libc::readlinkat(
                 dirfd,
-                cstr.as_ptr(),
+                cstr.as_ptr() as *const libc::c_char,
                 v.as_mut_ptr().cast(),
                 v.capacity() as size_t,
             ),
             None => libc::readlink(
-                cstr.as_ptr(),
+                cstr.as_ptr() as *const libc::c_char,
                 v.as_mut_ptr().cast(),
                 v.capacity() as size_t,
             ),
diff --git a/src/lib.rs b/src/lib.rs
index c4c0fa53..81896e7d 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -376,7 +376,7 @@ where
     F: FnOnce(*const libc::c_char) -> T,
 {
     match path {
-        Some(path) => path.with_nix_path(|p_str| f(p_str.as_ptr())),
+        Some(path) => path.with_nix_path(|p_str| f(p_str.as_ptr() as *const libc::c_char)),
         None => Ok(f(ptr::null())),
     }
 }
diff --git a/src/mount/linux.rs b/src/mount/linux.rs
index 3c271507..1e278766 100644
--- a/src/mount/linux.rs
+++ b/src/mount/linux.rs
@@ -119,7 +119,7 @@ pub fn mount<
                 crate::with_opt_nix_path(data, |d| unsafe {
                     libc::mount(
                         s,
-                        t.as_ptr(),
+                        t.as_ptr() as *const libc::c_char,
                         ty,
                         flags.bits(),
                         d as *const libc::c_void,
@@ -135,7 +135,7 @@ pub fn mount<
 /// Unmount the file system mounted at `target`.
 pub fn umount<P: ?Sized + NixPath>(target: &P) -> Result<()> {
     let res =
-        target.with_nix_path(|cstr| unsafe { libc::umount(cstr.as_ptr()) })?;
+        target.with_nix_path(|cstr| unsafe { libc::umount(cstr.as_ptr() as *const libc::c_char) })?;
 
     Errno::result(res).map(drop)
 }
@@ -145,7 +145,7 @@ pub fn umount<P: ?Sized + NixPath>(target: &P) -> Result<()> {
 /// See also [`umount`](https://man7.org/linux/man-pages/man2/umount.2.html)
 pub fn umount2<P: ?Sized + NixPath>(target: &P, flags: MntFlags) -> Result<()> {
     let res = target.with_nix_path(|cstr| unsafe {
-        libc::umount2(cstr.as_ptr(), flags.bits())
+        libc::umount2(cstr.as_ptr() as *const libc::c_char, flags.bits())
     })?;
 
     Errno::result(res).map(drop)
diff --git a/src/net/if_.rs b/src/net/if_.rs
index 5b6c2e77..ef8aaa39 100644
--- a/src/net/if_.rs
+++ b/src/net/if_.rs
@@ -17,7 +17,7 @@ pub type IflagsType = libc::c_longlong;
 /// Resolve an interface into an interface number.
 pub fn if_nametoindex<P: ?Sized + NixPath>(name: &P) -> Result<c_uint> {
     let if_index = name
-        .with_nix_path(|name| unsafe { libc::if_nametoindex(name.as_ptr()) })?;
+        .with_nix_path(|name| unsafe { libc::if_nametoindex(name.as_ptr() as *const libc::c_char) })?;
 
     if if_index == 0 {
         Err(Error::last())
diff --git a/src/sys/stat.rs b/src/sys/stat.rs
index c5854eec..c765847b 100644
--- a/src/sys/stat.rs
+++ b/src/sys/stat.rs
@@ -253,7 +253,7 @@ pub fn fstatat<P: ?Sized + NixPath>(
     let res = pathname.with_nix_path(|cstr| unsafe {
         libc::fstatat(
             dirfd,
-            cstr.as_ptr(),
+            cstr.as_ptr() as *const libc::c_char,
             dst.as_mut_ptr(),
             f.bits() as libc::c_int,
         )
diff --git a/src/unistd.rs b/src/unistd.rs
index 58ede6eb..169705cc 100644
--- a/src/unistd.rs
+++ b/src/unistd.rs
@@ -3611,7 +3611,7 @@ impl Group {
         // at `grp`.
         unsafe {
             Group::from_anything(|grp, cbuf, cap, res| {
-                libc::getgrnam_r(name.as_ptr(), grp, cbuf, cap, res)
+                libc::getgrnam_r(name.as_ptr() as *const libc::c_char, grp, cbuf, cap, res)
             })
         }
     }
