
---
 src/lib.rs | 4 ++++
 1 file changed, 4 insertions(+)

diff --git b/src/lib.rs a/src/lib.rs
index 71139385..a0a0372d 100644
--- b/src/lib.rs
+++ a/src/lib.rs
@@ -24,6 +24,10 @@
 #![cfg_attr(feature = "rustc-dep-of-std", no_core)]
 #![cfg_attr(libc_const_extern_fn_unstable, feature(const_extern_fn))]
 
+// ANDROID: Use std to allow building as a dylib.
+#[cfg(android_dylib)]
+extern crate std;
+
 #[macro_use]
 mod macros;
 
